### PR TITLE
Fix space invite issues

### LINKED ISSUE #3345: Space Invite by link not always displayed for public space
E.g join #community:matrix.org, there is no option to share the link

Also the option to share a link is shown for private spaces

### LINKED ISSUE #3346: Wrong copy in share space bottom sheet
Here it's not my space, and it's not just me at the moment

image

^ this is the copy for post space creation

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #3345
Fixes #3346 
+ use alias if there is one in share link (if not roomId with via params)
+ refactor using view model for bottomsheet

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/spaces/share/ShareSpaceBottomSheet.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                }
+            }
+        }
+    }
+
+    companion object {
+
+        fun show(fragmentManager: FragmentManager, spaceId: String, postCreation: Boolean = false): ShareSpaceBottomSheet {
+            return ShareSpaceBottomSheet().apply {
+                isCancelable = true
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/spaces/share/ShareSpaceBottomSheet.kt:120]: We cannot cancel this Dialog?

File: vector/src/main/java/im/vector/app/features/spaces/share/ShareSpaceViewEvents.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+
+package im.vector.app.features.spaces.share
+
+import im.vector.app.core.platform.VectorViewEvents
+
+sealed class ShareSpaceViewEvents : VectorViewEvents {
+    data class NavigateToInviteUser(val spaceId: String) : ShareSpaceViewEvents()
+    data class ShowInviteByLing(val permalink: String, val spaceName: String) : ShareSpaceViewEvents()
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/spaces/share/ShareSpaceViewEvents.kt:23]: Typo Ling -> Link

**Review by bmarty — APPROVED** [review]
LGTM, 2 small remarks

File: vector/src/main/java/im/vector/app/features/spaces/share/ShareSpaceBottomSheet.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/spaces/share/ShareSpaceBottomSheet.kt:120]: yes we can, i 'll remove as it's the default anyway

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/spaces/share/ShareSpaceBottomSheet.kt]: ha yes, `true != false` :)

**Pull request merged**